test/lsp.py

```diff
@@ -898,18 +898,28 @@ def main(self) -> int:
 
         return min(max(self.test_counter.failed, self.assertion_counter.failed), 127)
 
-    def setup_lsp(self, lsp: JsonRpcProcess, expose_project_root=True):
+    def setup_lsp(
+        self,
+        lsp: JsonRpcProcess,
+        expose_project_root=True,
+        analyze_all_files_in_project=False,
+        project_root_subdir=None
+    ):
         """
         Prepares the solc LSP server by calling `initialize`,
         and `initialized` methods.
         """
+        project_root_dir_uri_with_maybe_subdir = self.project_root_uri
+        if project_root_subdir is not None:
+            project_root_dir_uri_with_maybe_subdir = self.project_root_uri + '/' + project_root_subdir
         params = {
             'processId': None,
-            'rootUri': self.project_root_uri,
+            'rootUri': project_root_dir_uri_with_maybe_subdir,
             # Enable traces to receive the amount of expected diagnostics before
             # actually receiving them.
             'trace': 'messages',
-            'initializationOptions': {},
+            'initializationOptions': {
+            },
```

Comment:
Looks like an unnecessary change.
```suggestion
            'initializationOptions': {},
```